### PR TITLE
Started to convert more of MCServer to std::chrono

### PR DESCRIPTION
**Pull request by worktycho**

### CODE REVIEW

**Comment by madmaxoft** [comment]
Just out of curiosity, what is `sizeof(std::chrono::milliseconds)`? Is it an object small enough to pass by value?

**Comment by worktycho** [comment]
On any sensible implementation of the standard library it should just be a single 64 bit integer field. I have checked this is true for the libstdc++4.8. It was designed to be used in exactly this way.

**Comment by madmaxoft** [comment]
Sounds good enough.

**Comment by tigerw** [comment]
Presumably you've agreed not to use the user defined literals?

**Comment by worktycho** [comment]
I wanted to maximise the chance of the PR getting accepted. I might add them in a future PR.

File: src/World.h [hunk 1]
```diff
@@ -1051,13 +1049,13 @@ class cWorld :
 	cWorld(const AString & a_WorldName, eDimension a_Dimension = dimOverworld, const AString & a_OverworldName = "");
 	virtual ~cWorld();
 
-	void Tick(float a_Dt, int a_LastTickDurationMSec);
+	void Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_LastTickDurationMSec);
```
Comment (madmaxoft) [inline src/World.h]: Now the real question: What's the difference between `a_Dt` and `a_LastTickDurationMSec`?

Comment (worktycho) [inline src/World.h]: a_Dt is how long between the start of the last tick and this on, a_LastTickDurationMSec is how long the last tick took to execute. I'm not sure anything uses a_LastTickDurationMSec.

**Comment by worktycho** [comment]
Would someone mind reviewing this code?

File: src/World.cpp [hunk 2]
```diff
@@ -869,30 +865,27 @@ void cWorld::Tick(float a_Dt, int a_LastTickDurationMSec)
...
 
-	m_WorldAgeSecs  += (double)a_Dt / 1000.0;
-	m_WorldAge  = (Int64)(m_WorldAgeSecs  * 20.0);
+	m_WorldAge += a_Dt;
 
 	if (m_IsDaylightCycleEnabled)
 	{
 		// We need sub-tick precision here, that's why we store the time in seconds and calculate ticks off of it
-		m_TimeOfDaySecs += (double)a_Dt / 1000.0;
+		m_TimeOfDay += a_Dt;
```
Comment (madmaxoft) [inline src/World.cpp]: The comment above seems somewhat misleading now. Isn't `m_TimeOfDay` in msec?

**Comment by madmaxoft** [comment]
Apart from the comment issue this seems alright.

**Comment by worktycho** [comment]
I could reduce some of the chattyness by using user defined literals though I'm not sure what support for them is like.

**Pull request merged**